File: scripts/utils/count_matches.py
#!/usr/bin/env python3
import json
from collections import Counter
from pathlib import Path

input_path = Path(__file__).resolve().parents[2] / 'temp' / 't.json'
output_path = Path(__file__).resolve().parents[2] / 'temp' / 'match_counts.json'

if not input_path.exists():
    print(f'Input file not found: {input_path}')
    raise SystemExit(2)

with input_path.open('r', encoding='utf-8') as f:
    data = json.load(f)

counter = Counter()
for sample in data.get('samples', []):
    for m in sample.get('matches', []):
        key = str(m.get('match'))
        counter[key] += 1

sorted_items = sorted(counter.items(), key=lambda x: (-x[1], x[0]))

# Print
print('count  match')
for k, v in sorted_items:
    print(f'{v:5d}  {k}')

# Write JSON
with output_path.open('w', encoding='utf-8') as f:
    json.dump([{'match': k, 'count': v} for k, v in sorted_items], f, ensure_ascii=False, indent=2)

print(f'\nWrote {len(sorted_items)} items to {output_path}')

File: scripts/utils/count-matches.js
const fs = require("fs");
const path = require("path");

const inputPath = path.resolve(__dirname, "../../temp/t.json");
const outputPath = path.resolve(__dirname, "../../temp/match_counts.json");

async function main() {
  if (!fs.existsSync(inputPath)) {
    console.error(`Input file not found: ${inputPath}`);
    process.exit(2);
  }

  let raw;
  try {
    raw = await fs.promises.readFile(inputPath, "utf8");
  } catch (err) {
    console.error("Failed to read input file:", err.message);
    process.exit(2);
  }

  let data;
  try {
    data = JSON.parse(raw);
  } catch (err) {
    console.error("Failed to parse JSON:", err.message);
    process.exit(2);
  }

  const counts = new Map();
  const samples = Array.isArray(data.samples) ? data.samples : [];
  for (const sample of samples) {
    if (!Array.isArray(sample.matches)) continue;
    for (const m of sample.matches) {
      const key = String(m.match);
      counts.set(key, (counts.get(key) || 0) + 1);
    }
  }

  const sorted = Array.from(counts.entries())
    .map(([match, count]) => ({ match, count }))
    .sort((a, b) => b.count - a.count || a.match.localeCompare(b.match));

  // Print a readable list
  console.log("Match counts (sorted):");
  for (const { match, count } of sorted) {
    console.log(`${count.toString().padStart(4)}  ${match}`);
  }

  // Save JSON file for further inspection
  try {
    await fs.promises.writeFile(
      outputPath,
      JSON.stringify(sorted, null, 2),
      "utf8"
    );
    console.log(`\nWrote counts to ${outputPath}`);
  } catch (err) {
    console.error("Failed to write output file:", err.message);
    process.exit(2);
  }
}

if (require.main === module) {
  main().catch((err) => {
    console.error(err);
    process.exit(1);
  });
}

module.exports = { main };
